graph/cycle_detection.cpp

```diff
@@ -0,0 +1,117 @@
+#include <vector>
+#include <algorithm>
+#include <iostream>
+
+using namespace std;
```

Comment:
This is discouraged. Please strictly use `std::` for each of its members.

Reply:
Yeah completely forgot about that , i'll change it today